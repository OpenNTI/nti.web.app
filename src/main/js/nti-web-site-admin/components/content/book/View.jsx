import {Router, Route} from '@nti/web-routing';

import UserBookEnrollment from '../../user-book-enrollment';

import Overview from './overview';
import Reports from './reports';
import Roster from './roster';
import Frame from './Frame';

export default Router.for([
	Route({
		path: '/roster/:userID',
		component: UserBookEnrollment,
		props: {courseContext: true},
		frameless: true,
		getRouteFor (obj, context) {
			if (obj.MimeType === 'application/vnd.nextthought.userbundlerecord' && context === 'site-admin.courses.book-roster.list') {
				return `/roster/${obj.User.getID()}`;
			}

			return null;
		}
	}),
	Route({
		path: '/',
		component: Router.for([
			Route({ path: '/reports', component: Reports, name: 'site-admin.courses.book-reprots' }),
			Route({ path: '/roster', component: Roster, name: 'site-admin.courses.book-roster' }),
			Route({path: '/', component: Overview, name: 'site-admin.courses.book-overview'})
		], {frame: Frame})
	})
]);
