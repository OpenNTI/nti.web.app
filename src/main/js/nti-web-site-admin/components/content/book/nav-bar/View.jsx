import React from 'react';

import Card from '../../../common/Card';

import Identity from './Identity';
import Tabs from './Tabs';

export default function SiteAdminBookNavBar (props) {
	return (
		<Card className="site-admin-course-nav-bar">
			<Identity {...props} />
			<Tabs {...props} />
		</Card>
	);
}
