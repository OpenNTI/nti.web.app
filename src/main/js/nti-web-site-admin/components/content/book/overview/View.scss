@import '~nti-style-common';

.site-admin-book-overview {
	@include clearfix;

	.site-admin-active-days {
		width: 100%;
		margin-top: 5px;
	}

	.active-users {
		width: 100%;

		.dashboard-list-widget {
			margin: 0 0 10px;
		}
	}

	.active-times-widget {
		margin: 0 0 10px;
	}

	.site-admin-active-days {
		margin: 10px 0;
	}

	.site-admin-row {
		display: flex;

		.site-admin-card {
			width: 250px;
			height: 320px;

			.value {
				padding-top: 1rem;
			}
		}

		.active-users {
			height: 320px;

			.header {
				margin-top: 0;
				margin-bottom: 0;
			}
		}
	}
}
