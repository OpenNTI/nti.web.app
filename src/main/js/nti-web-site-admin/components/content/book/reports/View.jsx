import React from 'react';
import PropTypes from 'prop-types';
import { List } from '@nti/web-reports';

import Card from '../../../common/Card';


SiteAdminBooksReports.propTypes = {
	course: PropTypes.object
};
export default function SiteAdminBooksReports ({ course }) {
	return course ?
		(<Card><List context={course} /></Card>) :
		null;
}
