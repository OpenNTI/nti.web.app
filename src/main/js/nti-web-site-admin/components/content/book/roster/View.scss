@import '~@nti/style-common';

.site-admin-course-roster {
	.loading-mask {
		position: relative;
		min-height: 200px;
	}

	ul {
		li {
			box-shadow: 0 1px 0 0 var(--border-grey-light);

			a {
				text-decoration: none;
				display: block;
			}
		}
	}

	.controls {
		span {
			color: var(--primary-blue);
			margin-right: 10px;
			cursor: pointer;
		}
	}
}
