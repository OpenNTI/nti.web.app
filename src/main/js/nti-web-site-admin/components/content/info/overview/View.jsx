import PropTypes from 'prop-types';

import { AdminTools } from '@nti/web-course';

SiteAdminCourseOverview.propTypes = {
	course: PropTypes.object,
};
export default function SiteAdminCourseOverview({ course }) {
	return <AdminTools.Dashboard course={course} />;
}
