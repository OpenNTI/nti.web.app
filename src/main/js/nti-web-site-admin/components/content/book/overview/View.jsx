import './View.scss';
import React from 'react';
import PropTypes from 'prop-types';
import { Widgets } from '@nti/web-reports';

const { ActiveDays, ActiveUsers, ActiveTimes } = Widgets;

export default class SiteAdminBookOverview extends React.Component {
	static propTypes = {
		course: PropTypes.object
	}

	state = {}

	componentDidMount () {
		const {course} = this.props;

		if(!this.state.resp && course && course.hasLink('users')) {
			course.fetchLink('users').then(resp => {
				this.setState({resp});
			});
		}
	}

	render () {
		const {course} = this.props;

		return (
			<div className="site-admin-book-overview">
				<div className="site-admin-row">
					<div className="active-users">
						<ActiveUsers entity={course}/>
					</div>
				</div>
				<div className="active-days">
					<ActiveDays entity={course}/>
				</div>
				<div className="active-times">
					<ActiveTimes course={course}/>
				</div>
			</div>
		);
	}
}
