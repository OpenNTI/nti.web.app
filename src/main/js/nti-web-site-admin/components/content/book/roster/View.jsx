import './View.scss';
import React from 'react';
import PropTypes from 'prop-types';
import {List, Loading} from '@nti/web-commons';
import {decorate} from '@nti/lib-commons';
import {scoped} from '@nti/lib-locale';
import {LinkTo} from '@nti/web-routing';
import {searchable} from '@nti/web-search';

import ErrorMessage from '../../../common/ErrorMessage';
import Pager from '../../../common/Pager';
import Card from '../../../common/Card';

import Item from './Item';
import BookRosterStore from './BookRosterStore';

const DEFAULT_TEXT = {
	empty: 'No users have access to this book.',
	loadMore: 'Load More'
};
const t = scoped('nti-site-admin.courses.course.info.Roster', DEFAULT_TEXT);

class SiteAdminBookRoster extends React.Component {
	static propTypes = {
		store: PropTypes.object.isRequired,
		course: PropTypes.object,
		items: PropTypes.array,
		loading: PropTypes.bool,
		error: PropTypes.string,
		numPages: PropTypes.number,
		pageNumber: PropTypes.number
	}

	componentDidMount () {
		this.props.store.loadBook(this.props.course);
	}

	render () {
		const {loading, error, items, store, numPages, pageNumber} = this.props;

		return (
			<div className="site-admin-course-roster">
				{loading && (<div className="loading-mask"><Loading.Mask /></div>)}
				{error && (<ErrorMessage>{t('error')}</ErrorMessage>)}
				{!loading && items && this.renderItems(items)}
				{!loading && !error && <Pager store={store} numPages={numPages} pageNumber={pageNumber}/>}
			</div>
		);
	}

	renderItems (items) {
		if(items.length === 0) {
			return <div className="empty-state">{t('empty')}</div>;
		}

		return (
			<Card>
				<List.Unadorned>
					{items.map((item, index) => {
						return (
							<li key={index}>
								<LinkTo.Object object={item} context="site-admin.courses.book-roster.list">
									<Item item={item} />
								</LinkTo.Object>
							</li>
						);
					})}
				</List.Unadorned>
			</Card>
		);
	}

}

export default decorate(SiteAdminBookRoster, [
	searchable(),
	BookRosterStore.connect({
		loading: 'loading',
		items: 'items',
		error: 'error',
		numPages: 'numPages',
		pageNumber: 'pageNumber'
	})
]);
