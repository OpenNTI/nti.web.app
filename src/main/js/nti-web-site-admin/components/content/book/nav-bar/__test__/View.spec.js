import renderer, { act } from 'react-test-renderer';

import View from '../View';

/* eslint-env jest */
//TODO: make this more useful
describe.skip('Site admin user book nav bar test', () => {
	test('Basic render test', async () => {
		const book = {
			title: 'book1',
			getDefaultAssetRoot() {
				return 'testRoot';
			},
		};

		let cmp;
		act(() => {
			cmp = renderer.create(<View book={book} />);
		});

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
		cmp.unmount();
	});
});
