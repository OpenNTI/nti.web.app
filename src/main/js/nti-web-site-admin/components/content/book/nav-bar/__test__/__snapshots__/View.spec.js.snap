// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin user book nav bar test Basic render test 1`] = `
<div
  className="site-admin-card site-admin-course-nav-bar"
>
  <div
    className="site-admin-course-identity"
  >
    <img
      className="course-icon"
      src="contentpackage-landing-232x170.png?t=0"
    />
    <div
      className="info"
    >
      <div
        className="label"
      />
      <div
        className="title"
      >
        book1
      </div>
    </div>
  </div>
  <ul
    className="site-admin-tabs"
  >
    <li>
      <div
        className="nti-link-to-path"
        exact={true}
        to="./"
      >
        Overview
      </div>
    </li>
    <li>
      <div
        className="nti-link-to-path"
        to="./roster"
      >
        Roster
      </div>
    </li>
    <li>
      <div
        className="nti-link-to-path"
        to="./reports"
      >
        Reports
      </div>
    </li>
  </ul>
</div>
`;
